stubs/tensorflow/tensorflow/experimental/dtensor.pyi

```diff
@@ -0,0 +1,19 @@
+from _typeshed import Incomplete
+
+import numpy as np
+import numpy.typing as npt
+from tensorflow._aliases import IntDataSequence
+
+class Mesh:
+    def __init__(
+        self,
+        dim_names: list[str],
+        global_device_ids: npt.NDArray[np.uint] | IntDataSequence,
```

Comment:
Hmm would np.int also work? Would be a little surprised if it forced unsigned int.

Reply:
I suppose it would work as long as the values are not negative. The doc says:

> Values in this array should increment sequentially from 0.

Which to me implied no negative values, hence np.uint (looking at [the code](https://github.com/tensorflow/tensorflow/blob/2e86693cc76a2222750f7d807ed76b5672d1fc5c/tensorflow/dtensor/python/layout.py#L136), negative values would indeed not work).

Reply:
I would have leaned towards keeping a strict uint, but `IntDataSequence` already allows negative number anyways, so I'll add np.int to be consistent.

Reply:
Done in ebe2af975.